Allow sending records by sms to be disabled in config

sbujam opened the pull request:
Currently, sending record results by sms ("Text this" feature) is
always shown in the record page, even though it may not be applicable
to some installations.

Although still left enabled by default, with this patch, the admin
has the option to suppress the sms link in the record page, if needed.

The [Mail] section in main config file has been chosen to hold the new
parameter because sending sms messages uses the underline mailer
infrastructure.

demiankatz commented:
Thanks! This looks reasonable to me. I'll test and likely merge this either later today or early next week.

On a related note, there is currently no way to disable email support. Do you think we should work on another PR to make this configurable too, as it is just about the last remaining hard-coded feature? Controlling email is a bit more complicated than SMS because it shows up in more places (search results, record pages, cart, bulk action) and perhaps we want the ability to turn it off in some places but not others. Certainly not something we need to spend time on if nobody needs it, but just a thought.

demiankatz commented:
Okay, I have manually merged this to master with a couple of small additions:

- I made sure that the controller throws a ForbiddenException if a user tries to manually access the SMS URL when it is disabled -- this is a little more complete than simply hiding the link.

- I added support in the jquerymobile theme. This will be deprecated very soon, but since it still exists, it seemed worth keeping it in sync, since the change was trivial.

Thanks again for working on this!

sbujam commented:
Thank **you** Demian,

> I made sure that the controller throws a ForbiddenException if a user tries to manually access the SMS URL when it is disabled -- this is a little more complete than simply hiding the link.

I was too thinking that this may occur but decided not to take any precautions because even if one tries to use the /SMS suffix for a record, without proper configuration for the carrier, nothing will be done :)
But then again, you are right to make sure that an Exception is thrown just in case...

> I added support in the jquerymobile theme. This will be deprecated very soon, but since it still exists, it seemed worth keeping it in sync, since the change was trivial.

Whoops, I completely forgot the mobile theme! (we're still using jquerymobile in our production system and we removed the appropriate lines the hard way)

demiankatz commented:
No problem. Are you okay with the fact that jquerymobile will soon be removed, or do you still find it valuable? (Right now we're on track to delete it this week unless somebody objects, since it costs a lot of time to maintain, but most people have stopped using it since the responsive features of bootstrap3 meet most of the same needs).

sbujam commented:
Yes, i've read it in the roadmap. To be honest I kinda liked the minimal (and clean) interface of jquerymobile, as opposed to having a resized (but full-featured) mobile site, but I also understand the additional maintenance effort required to support another theme. So, baseline is: there might be some initial complaints from our users (as always with _any_ change), but we'll get used to it :)
After all modern phones have higher resolutions and can easily display the fully featured site without problems!

demiankatz commented:
Yes, I agree that there are some nice features of jquerymobile. Of course, I do plan to keep it alive in the form of a legacy branch, so if there's enough demand, you're always free to merge it back in locally and contribute updates back to the branch.
